pkg/provider/data_source_cluster.go

```diff
@@ -288,9 +290,32 @@ func (c *clusterDataSource) Schema(ctx context.Context, req datasource.SchemaReq
 				MarkdownDescription: "Earliest backup recover time.",
 				Computed:            true,
 			},
+			"maintenance_window": schema.SingleNestedAttribute{
+				Computed:            true,
+				MarkdownDescription: "Custom maintenance window.",
+				Attributes: map[string]schema.Attribute{
+					"is_enabled": schema.BoolAttribute{
+						MarkdownDescription: "Is maintenance window enabled.",
+						Computed:            true,
+					},
+					"start_day": schema.Float64Attribute{
```

Comment:
What's the reason that we use Float64Attribute instead of Int64Attribute here? (I checked all the `start_day` implementations, it's `Float64Attribute` everywhere, btw.) 
It's good to have the same implementation everywhere, yet I guess we can use Int64 here instead of Float64. 
WDYT, @wai-wong-edb ?

Reply:
Yes this is a previous problem where we are mixing the terraform models with the API models. The terraform models should use int64 and the API models need float64. I can change it to separate the terraform model and API model and the terraform model will convert the int64 to float64. Maybe I could do this for PGD too in this PR as it will affect PGD too